src/utils/preferences.js

```diff
@@ -0,0 +1,48 @@
+import React, { useContext } from 'react';
+import { useLocalStorageState } from './utils';
+import { useInterval } from './useInterval';
+import { useConnection } from './connection';
+import { useWallet } from './wallet';
+import { useAllMarkets, useTokenAccounts } from './markets';
+import { settleAllFunds } from './send';
+
+const ConnectionContext = React.createContext(null);
+
+export function PreferencesProvider({ children }) {
+  const [autoSettleEnabled, setAutoSettleEnabled] = useLocalStorageState(
+    'autoSettleEnabled',
+    false,
+  );
+
+  const [tokenAccounts] = useTokenAccounts();
+  const marketList = useAllMarkets();
```

Comment:
We can instead only run this on the markets for which the user has settleable OpenOrders accounts -- see `useUnmigratedOpenOrdersAccounts`